domain/modelupgrade/params.go

```diff
@@ -0,0 +1,17 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package modelupgrade
+
+import (
+	"github.com/juju/juju/core/semversion"
+)
+
+// UpgradeModelParams holds the parameters used to upgrade a model.
+type UpgradeModelParams struct {
+	ControllerModelVersion semversion.Number
```

Comment:
I don't think the caller should be supplying this information to us. We have the authoritative source for this and we should do this work.

Reply:
The service operates on the model database of the model being upgraded.  We don't have access to the controller model in the service.